duckduckgo/macos-browser
# MainMenuManager refactored to MainMenu, subclass of NSMenu

Task/Issue URL: https://app.asana.com/0/1199230911884351/1199874167124911/f
CC: @samsymons 

**Description**:
Bookmarks section of the main menu needs to be adjusted every time bookmarks change. Instead of managing the menu from other class/singleton MainMenuManager, let's subclass NSMenu and assign the main menu class in the storyboard 
**Advantages**
- Single responsibility principle - MainMenu can dynamically manage itself instead of other instances/classes managing it
- No lifecycle difference of NSMenu instance vs singleton(MainMenuManager)


**Steps to test this PR**:
1. Run the browser, make sure the Help menu item contains the "Send feedback" item.
2. Remove FEEDBACK from compile Active Compilation Conditions. Run the browser again. "Send feedback" item shouldn't be there


---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
